### PR TITLE
chore: migrate project group resource and datasource to framework

### PR DESCRIPTION
**Pull request by benPearce1**

### CODE REVIEW

**Review by IsaacCalligeros95 — COMMENTED** [review]
Should the octopusdeploy datasource and resource be removed?

File: octopusdeploy_framework/resource_project_group.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+		Attributes: schemas.GetProjectGroupResourceSchema(),
+	}
+}
+
+func (r *projectGroupTypeResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	r.Config = ResourceConfiguration(req, resp)
+}
+
+func (r *projectGroupTypeResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data *schemas.ProjectGroupTypeResourceModel
```
Comment (IsaacCalligeros95) [inline octopusdeploy_framework/resource_project_group.go:37]: My lack of go understanding will shine through here. Is there a reason this is a pointer while the other read, delete update methods are not?

**Review by IsaacCalligeros95 — APPROVED** [review]
LGTM, one question on the pointer in Create vs other methods.

File: octopusdeploy_framework/resource_project_group.go (hunk 1 above)
Comment (benPearce1) [reply octopusdeploy_framework/resource_project_group.go:37]: yeah, I think you are right, I will fix that

**Pull request merged**